builder/googlecompute/block_device_test.go

```diff
@@ -406,6 +415,28 @@ func TestGenerateDiskAttachment(t *testing.T) {
 				DeviceName:        "packer-test",
 			},
 		},
+		{
+			name: "basic persistent disk with source image",
+			config: BlockDevice{
+				SourceImage: "projects/p/global/images/family/f",
+				VolumeType:  "pd-standard",
+				DeviceName:  "packer-test",
+				DiskName:    "packer-test-disk",
+			},
+			expectval: &compute.AttachedDisk{
+				AutoDelete:        true,
+				Boot:              false,
+				DiskEncryptionKey: &compute.CustomerEncryptionKey{},
+				Interface:         "SCSI",
+				Mode:              "READ_WRITE",
+				Type:              "PERSISTENT",
+				DeviceName:        "packer-test",
+				InitializeParams: &compute.AttachedDiskInitializeParams{
+					DiskName:    "packer-test-disk",
+					SourceImage: "projects/p/global/images/family/f",
```

Comment:
I'm surprised we need to define the `SourceImage` for both the `BlockDevice` and the `AttachedDisk` objects, is this required?